lib/chef/resource/yum_package.rb

```diff
@@ -30,13 +30,13 @@ class YumPackage < Chef::Resource::Package
       property :flush_cache, Hash, default: { before: false, after: false }, coerce: proc { |v|
         # TODO these append rather than set. This is probably wrong behavior, but we're preserving it until we know
         if v.is_a?(Array)
-          v.each { |arg| flush_cache[arg] = true }
-          flush_cache
+          x = {}
+          v.each { |arg| x[arg] = true }
```

Comment:
collect won't work cause need a hash, but each_with_object is fine (and the same)